### DISCUSSION TITLE
Does Portman provides automatic fuzzing for "happy" test paths.

### DISCUSSION THREAD

**Discussion by PaulCluver** [apideck-libraries/portman#505] [Q&A] [answered]
Hi there, 

The fuzzing for the "unhappy" test paths works beautifully. Is there an option to "fuzz" for happy test paths. In other words, automatically removing object or key:value pairs  from the generated request that are NOT required? 

Kind regards,
Paul

BTW: Portman has truly revolutionalized how I do API testing. It is a fantastic tool.

**Comment by thim81** [comment]
@PaulCluver 

I'll have to review the code, but I think the Fuzzy testing is mostly focused on finding "unhappy" situations.
Could you explain to me why you would want to remove non-required properties?

**Reply by PaulCluver** [reply]
Thanks for getting back to me. My idea was sparked by the following from documentation at https://github.com/apideck-libraries/portman/tree/main/examples/testsuite-fuzzing-tests

![image](https://github.com/apideck-libraries/portman/assets/5680263/c3f0c007-d4f7-4f09-b26e-4d2b6b5cfa3b)

Instead of me having to configure all happy path variations, I was hoping that these could simply be generated. I am dealing with a very large request body so tracking down all possible 2xx variations which need to be exercised is somewhat laborious. 

Another example is with a variation test that I have written. This test removes a non-required portion of the request body using the below: 

![image](https://github.com/apideck-libraries/portman/assets/5680263/bcfecedf-85d0-48a9-9928-08c57ec292bc)

Submitting this request, as the removed property is not required, should result in a 2xx response. Instead, I get a 400 - Bad Request. While I have captured this as an issue, I am concerned that there may be many more. This is why I was hoping to be able to generate all possible permutations of a request where I remove non-required properties.

I hope that this provides some context. 

Kind regards,
Paul.

**Reply by thim81** [reply, marked as answer]
@PaulCluver 

I think it would be possible to develop the happy path option for removing "non-required" properties. It has been a while so we would need some time to dive in again.

**Reply by PaulCluver** [reply]
Hi Tim, 

Thanks so much for considering this as a an option. 

Kind regards,
Paul.

**Reply by thim81** [reply]
hi @PaulCluver 

Apologies, but we haven't found the time to look into this.
